### PR TITLE
docs(security): content security policies

### PR DESCRIPTION
**Pull request by sean-perkins**
Updates the security guide to include information about content security policies and how to assign the nonce value for Ionic web component projects.

Includes additional information for Angular 16 projects.

Preview link: https://ionic-docs-git-sp-csp-nonce-ionic1.vercel.app/docs/techniques/security#content-security-policies-csp

### CODE REVIEW

File: docs/techniques/security.md [hunk 1]
```diff
@@ -160,3 +160,53 @@ export const ToastExample: React.FC = () => {
 </TabItem>
 </Tabs>
 ````
+
+## Content Security Policies (CSP)
+
+A Content Security Policy (CSP) is a security mechanism that helps protect web applications against certain types of attacks, such as cross-site scripting (XSS) and data injection. It is implemented through an HTTP header that instructs the browser on which sources of content, such as scripts, stylesheets, and images, are allowed to be loaded and executed on a web page.
```
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply docs/techniques/security.md:166]: I think we should link out to MDN so devs can learn more

File: docs/techniques/security.md [hunk 2]
```diff
@@ -160,3 +160,53 @@ export const ToastExample: React.FC = () => {
...
+
+:::
+
+For more information on how to use CSPs with Stencil web components, see the [Stencil documentation](https://stenciljs.com/docs/csp-nonce).
+
+### Angular
+
+Starting in Angular 16, Angular provides two options for setting the nonce value.
+
+1. Set the `ngCspNonce` attribute on the root application element as `<app ngCspNonce="randomNonceGoesHere"></app>`. Use this approach if you have access to server-side templating that can add the nonce both to the header and the `index.html` when constructing the response.
```
[the remark this answers is not in the document]
Comment (sean-perkins) [reply docs/techniques/security.md:203]: Are you thinking replacing the content entirely with a link out to the Angular docs?

[the remark this answers is not in the document]
Comment (liamdebeasi) [reply docs/techniques/security.md:203]: No I was thinking of having the link there for devs who want to know more about the `ngCspNonce` attribute in addition to the copy you have

**Review by liamdebeasi — APPROVED** [review]
Good to go once my other comment is addressed

File: docs/techniques/security.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply docs/techniques/security.md:203]: Discussed offline and Sean does link to the main CSP section below which I think should be sufficient here.

[the remark this answers is not in the document]
Comment (sean-perkins) [reply docs/techniques/security.md:203]: We spoke on Slack - the link below this section will suffice for now 👍

**Pull request merged**